dor/providers/process_basic_image.py

```diff
@@ -379,6 +252,152 @@ def create_package_resource(
     return resource
 
 
+@dataclass
+class Operation(ABC):
+    accumulator: Accumulator
+
+    @abstractmethod
+    def run(self) -> None:
+        raise NotImplementedError
+
+
+@dataclass
+class Command:
+    operation: Type[Operation]
+    kwargs: dict[str, Any]
+
+
+@dataclass
+class Input:
+    file_path: Path
+    commands: list[Command]
+
+
+@dataclass
+class CopySource(Operation):
+    image_path: Path
+
+    @staticmethod
+    def copy_source_file(source_path: Path, destination_path: Path) -> None:
+        shutil.copyfile(source_path, destination_path)
+
+    def run(self) -> None:
+        try:
+            source_tech_metadata = TechnicalMetadata.create(self.image_path)
+        except JHOVEDocError:
+            return None
+
+        file_info = FileInfo(
+            identifier=self.accumulator.file_set_identifier.identifier,
+            basename=self.accumulator.file_set_identifier.basename,
+            uses=[UseFunction.source, UseFormat.image],
+            mimetype=source_tech_metadata.mimetype.value,
+        )
```

Comment:
Probably not something we should resolve in this PR, but this operation assumes that the file is an image (see 293), and the parameter is `image_path` (though that's easy enough to change).

We're using `CopySource` in the `inputs` loop, but this won't work for the transcription files. How should we handle this? Should we rely on the mimetype output from JHOVE? Or create a separate operation?

Reply:
Have the technical metadata return its `UseFormat`? That's the only part of this that doesn't work for non-image files, I think.